tasks/tests/integration/test_upload_e2e.py

```diff
@@ -0,0 +1,194 @@
+import json
+from functools import partial
+from typing import Iterable
+from uuid import uuid4
+
+import pytest
+from shared.reports.types import ReportLine
+from shared.yaml import UserYaml
+from sqlalchemy.orm import Session
+
+from database.tests.factories import CommitFactory, RepositoryFactory
+from services.archive import ArchiveService
+from services.report import ReportService
+from tasks.tests.unit.test_upload_task import FakeRedis
+from tasks.tests.utils import hook_session, run_tasks
+from tasks.upload import upload_task
+
+
+# TODO: instead of `mock_redis`, can we use the real thing?
+@pytest.fixture
+def mock_redis(mocker):
+    m = mocker.patch("services.redis._get_redis_instance_from_url")
+    redis_server = FakeRedis(mocker)
+    m.return_value = redis_server
+    yield redis_server
+
+
+def write_raw_upload(
+    redis: FakeRedis,
+    archive_service: ArchiveService,
+    repoid: int,
+    commitid: str,
+    contents: bytes,
+):
+    report_id = uuid4().hex
+
+    written_path = archive_service.write_raw_upload(commitid, report_id, contents)
+
+    redis_key = f"uploads/{repoid}/{commitid}"
+    uploads = redis.lists.setdefault(redis_key, [])
+    uploads.append(json.dumps({"reportid": report_id, "url": written_path}))
+
+
+def lines(lines: Iterable[tuple[int, ReportLine]]) -> list[tuple[int, int]]:
+    return list(((lineno, line.coverage) for lineno, line in lines))
+
+
+@pytest.mark.integration
+@pytest.mark.django_db()
+def test_full_upload(
+    dbsession: Session,
+    mocker,
+    mock_redis,
+    mock_repo_provider,
+    mock_storage,
+    mock_configuration,
+):
+    hook_session(dbsession, mocker)
+
+    # to not close the session after each task
+    mocker.patch("tasks.base.BaseCodecovTask.wrap_up_dbsession")
+    # avoid some calls that are failing for unknown reasons:
+    mocker.patch("tasks.upload.UploadTask.possibly_setup_webhooks", return_value=True)
+    mocker.patch(
+        "tasks.upload.fetch_commit_yaml_and_possibly_store", return_value=UserYaml({})
+    )
+    mocker.patch(
+        "tasks.upload.get_repo_provider_service",
+        return_value=mock_repo_provider,
+    )
+    mocker.patch(
+        "tasks.upload_processor.get_repo_provider_service",
+        return_value=mock_repo_provider,
+    )
+    mocker.patch(
+        "tasks.notify.get_repo_provider_service",
+        return_value=mock_repo_provider,
+    )
+    # for some reason, the `archive` down below still does not contain the files+sessions file?
```

Comment:
I suppose the files+sessions is only created by the `UploadProcessor` task, and maybe that haven't run at this point?

If the raw upload is missing then maybe...
I recently had similar issues in tests where I wrote things to archive and I had to patch this function to get it to work:
```
mocker.patch(
   "services.bundle_analysis.comparison.get_appropriate_storage_service",
   return_value=mock_storage,
)
```